axelor-admin/src/main/java/com/axelor/apps/base/service/app/DataBackupCreateService.java

```diff
@@ -175,31 +179,87 @@ public File create(Integer fetchLimit) {
   /* Get All MetaModels */
   private List<MetaModel> getMetaModels() {
     String filterStr =
-        "self.packageName NOT LIKE '%meta%' AND self.packageName !='com.axelor.studio.db' AND self.name!='DataBackup' AND self.tableName NOT LIKE 'BASE_APP%'";
+        "self.packageName NOT LIKE '%meta%' AND self.packageName !='com.axelor.studio.db' AND self.name!='DataBackup'";
     List<MetaModel> metaModels = metaModelRepo.all().filter(filterStr).fetch();
     metaModels.add(metaModelRepo.findByName(MetaFile.class.getSimpleName()));
     metaModels.add(metaModelRepo.findByName(MetaJsonField.class.getSimpleName()));
     return metaModels;
   }
 
+  @SuppressWarnings("unchecked")
+  private Map<String, List<String>> getSubClassesMap() {
+    List<MetaModel> metaModels = getMetaModels();
+    Map<String, List<String>> subClassMap = new HashMap<String, List<String>>();
+    List<String> subClasses;
+    Class<Model> superClass, subClass;
+    for (MetaModel metaModel : metaModels) {
+      subClasses = new ArrayList<>();
+      try {
+        superClass = (Class<Model>) Class.forName(metaModel.getFullName());
+        for (MetaModel metaModelItem : metaModels) {
+          subClass = (Class<Model>) Class.forName(metaModelItem.getFullName());
+          if (subClass.getSuperclass() == superClass) {
+            subClasses.add(metaModelItem.getName());
+          }
+        }
+      } catch (Exception e) {
+      }
+      if (subClasses != null && subClasses.size() > 0) {
+        subClassMap.put(metaModel.getFullName(), subClasses);
+      }
+    }
+    return subClassMap;
+  }
+
   /* Get All Data of Specific MetaModel */
-  private List<Model> getMetaModelDataList(MetaModel metaModel, int start, Integer fetchLimit) {
+  private List<Model> getMetaModelDataList(
+      MetaModel metaModel, int start, Integer fetchLimit, List<String> subClasses)
+      throws ClassNotFoundException {
+    String query = "";
+    if (subClasses != null && subClasses.size() > 0) {
+      String whereStr = "";
+      for (String subClassName : subClasses) {
+        whereStr += "id NOT IN (select id from " + subClassName + ")";
+        if (whereStr.length() > 0 && !subClassName.equals(subClasses.get(subClasses.size() - 1))) {
+          whereStr += " AND ";
+        }
```

Comment:
Instead of checking index of last subClass , you can just add "AND" before condition everytime except first time where its empty.